src/codegen/mod.rs

```diff
@@ -2949,8 +2957,15 @@ impl CodeGenerator for Function {
             vis: ast::Visibility::Public,
         };
 
-        let item = ForeignModBuilder::new(signature.abi()
-                .expect("Invalid or unknown ABI for function!"))
+        let abi = match signature.abi() {
+            Abi::Known(abi) => abi,
+            Abi::Unknown(unknown_abi) => {
+                panic!("Invalid or unknown abi {:?} for function {:?} ({:?})",
+                       unknown_abi, canonical_name, self);
```

Comment:
It is not, one prints canonical name and a `Function` type. The other doesn't. I expected to add an `unwrap_known_abi` method, but that wouldn't address #727